main.sh

```diff
@@ -91,12 +92,27 @@ trap cleanup_before_exit EXIT
 
 # Translate usage string -> getopts arguments, and set $arg_<flag> defaults
 while read line; do
-  opt="$(echo "${line}" |awk '{print $1}' |sed -e 's#^-##')"
+  # fetch single character version of option sting
+  opt="$(echo "${line}" |awk '{print $1}' |sed -e 's#^-\(.\)#\1#')"
```

Comment:
may as well set opt to just the first character of the first field... Some error checking here could be nice